cnxarchive/database.py

```diff
@@ -112,6 +113,36 @@ def initdb(settings):
             for schema_part in get_schema():
                 cursor.execute(schema_part)
 
+            """If virtualenv is active, use that for postgres"""
+            if hasattr(sys, 'real_prefix'):
+                import os
+                virtualenv_path = os.environ['VIRTUAL_ENV']
+
+                cursor.execute("SELECT current_database();")
+                db_name = cursor.fetchone()[0]
+
+                cursor.execute("""SELECT schema_name FROM
+                                  information_schema.schemata
+                                  WHERE schema_name = 'venv';""")
+                schema_exists = cursor.fetchall()
+
+                if not schema_exists:
+                    cursor.execute("CREATE SCHEMA venv")
+                    cursor.execute("ALTER DATABASE \"{}\" SET "
+                                   "session_preload_libraries ="
+                                   "'session_exec'".format(db_name))
+                    cursor.execute("ALTER DATABASE \"{}\" SET "
+                                   "session_exec.login_name = "
+                                   "'venv.activate_venv'".format(db_name))
+                    cursor.execute("""CREATE FUNCTION venv.activate_venv()
+                                   RETURNS void LANGUAGE plpythonu AS $_$
+                                   import os
+                                   activate_this =
+                                   os.path.join('{}/bin/activate_this.py')
+                                   exec(open(activate_this).read(),
```

Comment:
That would lock the initialized DB to the particular venv it was initialized from, then, right? So, effectively lock the trigger code and anything else imported to that venv.